src/timer.rs

```diff
@@ -0,0 +1,70 @@
+trait Timer: Sized {
+    fn start(milliseconds: u32) -> Self;
+    fn has_elapsed(self) -> Option<Self>;
+}
+
+use libtock_core::syscalls;
+
+// see: https://github.com/tock/tock/blob/master/doc/syscalls/00000_alarm.md
+mod command_nr {
+    pub const _IS_DRIVER_AVAILABLE: usize = 0;
+    pub const GET_CLOCK_FREQUENCY: usize = 1;
+    pub const GET_CLOCK_VALUE: usize = 2;
+    pub const _STOP_ALARM: usize = 3;
+    pub const _SET_ALARM: usize = 4;
+}
+const DRIVER_NUMBER: usize = 0x00000;
+
+struct LibtockAlarmTimer {
+    end_tick: usize,
+}
+
+fn wrapping_add_u24(lhs: usize, rhs: usize) -> usize {
+    lhs.wrapping_add(rhs) & 0xffffff
+}
+fn wrapping_sub_u24(lhs: usize, rhs: usize) -> usize {
+    lhs.wrapping_sub(rhs) & 0xffffff
+}
+
+impl Timer for LibtockAlarmTimer {
+    fn start(milliseconds: u32) -> Self {
+        let clock_frequency =
+            syscalls::command(DRIVER_NUMBER, command_nr::GET_CLOCK_FREQUENCY, 0, 0).unwrap_or(0);
```

Comment:
Mmmm... I'm not sure we should `unwrap_or(0)`. We should either `unwrap()` (or `flex_unwrap()` in that case), or propagate the error. Ideally we should propagate the error, but given that we will probably just unwrap it at the call site, it's probably alright to unwrap here (and improve later if needed).

Same below.

Reply:
I cannot use flex_unwrap because syscalls::command uses CommandError rather than TockError. I did the same thing as what is done in flex_unwrap, which is ok().unwrap()